frontend/packages/component-library/src/cms/header/Header.tsx

```diff
@@ -130,7 +130,20 @@ const Header: React.FC<HeaderProps> = ({
       try {
         const data = await fetchUserSignedInStatus(studioBaseUrl);
         if (data) {
-          const renderState = data.is_signed_in ? 'signedIn' : 'signedOut';
+          const isSignedIn = data.is_signed_in;
+
+          if (isSignedIn) {
+            const paths = window.location.pathname?.split('/').filter(Boolean);
+
+            if (paths.length === 0 || paths.length === 1) {
+              // If the user is signed in and on the home page, redirect to dashboard
+              // If parts.length === 0, then they are on code.org
+              // If parts.length === 1, then they are on a localized home page path like code.org/en-US
+              window.location.href = `${studioBaseUrl}`;
```

Comment:
This adds an interesting complication to any potential use of the `Header` component on dashboard itself ... but I think we can solve that problem later?

Reply:
When we refresh the components, I think we'll have to restructure these components to be more agnostic of the platform it is on. I agree that the implementation is highly specific to the marketing site right now and should be generalized at a later point